static/src/js/app/vuex/actions.js

```diff
@@ -39,13 +46,34 @@ export default {
   [FETCH_TEXT]: ({ commit }, { id }) => api
     .fetchText(id, (data) => commit(FETCH_TEXT, data.data))
     .catch(logoutOnError(commit)),
-  [FETCH_VOCAB_LISTS]: ({ commit, state }) => {
-    api
-      .fetchVocabLists(state.text.lang, (data) => commit(FETCH_VOCAB_LISTS, data.data))
+  [FETCH_VOCAB_LIST]: async ({ commit }, { vocabListId }) => {
+    const { data } = await api
+      .fetchVocabList(vocabListId)
+      .catch(logoutOnError(commit));
+    commit(UPDATE_VOCAB_LIST, data.data);
```

Comment:
Just thinking for future improvements we could alert the user if there is an error fetching the `vocab list`

Reply:
Yeah, it seems like a different way of surfacing errors might be a good thing to set up, maybe in a v1.1